test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,444 @@
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+    private void performFailoverTest() throws InterruptedException {
+
+        long currentWriteCount1;
+        long currentReadCount1;
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count without any failover {}", currentReadCount1);
+        log.info("Write count without any failover {}", currentWriteCount1);
+
+        int sleepTime;
+
+        //check reads and writes after some random time
+        sleepTime = new Random().nextInt(50000) + 3000;
+        log.info("Sleeping for {} ", sleepTime);
+        Thread.sleep(sleepTime);
+
+        long currentWriteCount2;
+        long currentReadCount2;
+
+        currentWriteCount2 = eventWriteCount.get();
+        currentReadCount2 = eventReadCount.get();
+
+        log.info("Read count without any failover after sleep before scaling  {}", currentReadCount2);
+        log.info("Write count without any failover after sleep before scaling {}", currentWriteCount2);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(60000);
+        log.info("Scaling down SSS instances from 3 to 2");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count after SSS failover after sleep  {}", currentReadCount1);
+        log.info("Write count after SSS failover after sleep {}", currentWriteCount1);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount1 > currentWriteCount2);
+        //ensure reads are happening
+        assertTrue(currentReadCount1 > currentReadCount2);
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(60000);
```

Comment:
writes and reads should continue to happen even with controller failover.
since you are doing txn writes, make sure that while controller instance is scaled down, we are still able to commit transactions (ensure that you dont just call commit, but check txn status and ensure it reaches committed state)